sylvia/tests/querier.rs

```diff
@@ -0,0 +1,125 @@
+use cosmwasm_std::{Addr, Response, StdError, StdResult};
+use cw_storage_plus::Item;
+use serde::{Deserialize, Serialize};
+use sylvia::types::{ExecCtx, InstantiateCtx, QueryCtx};
+use sylvia::{contract, interface};
+
+#[derive(
+    Serialize, Deserialize, Clone, PartialEq, Eq, sylvia::schemars::JsonSchema, Debug, Default,
+)]
+pub struct CountResponse {
+    pub count: u64,
+}
+
+pub mod counter {
+    use super::*;
+
+    #[interface]
+    pub trait Counter {
+        type Error: From<StdError>;
+
+        #[msg(query)]
+        fn count(&self, ctx: QueryCtx) -> StdResult<CountResponse>;
+
+        #[msg(exec)]
+        fn copy_count(&self, ctx: ExecCtx, contract_addr: Addr) -> StdResult<Response>;
+
+        #[msg(exec)]
+        fn set_count(&self, ctx: ExecCtx, new_count: u64) -> StdResult<Response>;
+    }
+
+    #[contract]
+    #[messages(counter as Counter)]
+    impl Counter for CounterContract {
+        type Error = StdError;
+
+        #[msg(query)]
+        fn count(&self, ctx: QueryCtx) -> StdResult<CountResponse> {
+            let count = self.count.load(ctx.deps.storage)?;
+            Ok(CountResponse { count })
+        }
+
+        #[msg(exec)]
+        fn set_count(&self, ctx: ExecCtx, new_count: u64) -> StdResult<Response> {
+            self.count.save(ctx.deps.storage, &new_count)?;
+            Ok(Response::new())
+        }
+
+        #[msg(exec)]
+        fn copy_count(&self, ctx: ExecCtx, contract_addr: Addr) -> StdResult<Response> {
+            let other_count = counter::Remote::new(contract_addr)
+                .querier(&ctx.deps.querier)
+                .count()?
+                .count;
```

Comment:
I like what it is called. However, please, move to create `Remote::new(contract_addr)` to the instantiation and store it in the contract. This way, we show a crucial use case we want to support - keeping `Remote` as the contract state and later reaching it. Here you would just load `Remote` from the state.